GH-4362 Add syncing status

i-am-chitti opened the pull request:
This PR adds a `syncing` status on frontend.

Issue
- #4362

FelixMalfait commented on packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsSynchronizationStatus.tsx:
```diff
@@ -1,27 +1,46 @@
+  syncStatus: MessageChannelSyncStatus;
+};
+
+const getColor = (syncStatus?: MessageChannelSyncStatus | null) => {
```
I've just noticed we defined colors and label in message-channel object metadata already. Maybe we want to get theme from here?

```
  let objectMetadataItem = useRecoilValue(
    objectMetadataItemFamilySelector({
      objectName: CoreObjectNameSingular.MessageChannel
      objectNameType: 'singular',
    }),
  );
  ```
  Then get object from field and...

```
  const selectedOptions = field.metadata.options.filter(
    (option) => fieldValues?.includes(option.value),
  );
 ```
 
 What do you think @lucasbordeau? Would you recommend this approach to access the option values?

charlesBochet replied:
Yes, we should use the options here. you can get the objectMetadataItem using useObjectMetadataItem hook directly

i-am-chitti replied:
@FelixMalfait,
I see options are present here - 
<img width="1199" alt="image" src="https://github.com/twentyhq/twenty/assets/60139930/6aaa69a9-867d-4ae8-96d5-7346cfd7fbb3">

FelixMalfait reviewed (commented):
Great! Just a small comment to avoid duplicating source of truth for colors

i-am-chitti commented:
@FelixMalfait, @charlesBochet, Addressed the feedback. Have a re-review.
One thing to note is that, on my local, I'm always getting syncStatus as null - 
<img width="346" alt="image" src="https://github.com/twentyhq/twenty/assets/60139930/258f8dbd-1acf-4533-9268-c1c9fa587d79">
when I logged https://github.com/twentyhq/twenty/blob/main/packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsMessageChannelsListCard.tsx#L53 with the new changes as in the PR. 

How can I make this in other status for testing?

charlesBochet commented on packages/twenty-front/src/modules/object-metadata/hooks/useGetSyncStatusMetadata.ts:
```diff
@@ -0,0 +1,25 @@
+import { CoreObjectNameSingular } from '@/object-metadata/types/CoreObjectNameSingular';
+import { isUndefinedOrNull } from '~/utils/isUndefinedOrNull';
+
+export const useGetSyncStatusMetadata = () => {
```
this should live in calendar front module (or in modules/settings/accounts) (it's not something generic like object-metadata)
other than than it looks good!

FelixMalfait reviewed (approved):
Thanks @i-am-chitti great contribution as always!
